editor/editor_help.cpp

```diff
@@ -225,35 +272,54 @@ void EditorHelp::_add_type(const String &p_type, const String &p_enum) {
 	bool is_enum_type = !p_enum.is_empty();
 	bool can_ref = !p_type.contains("*") || is_enum_type;
 
-	String t = p_type;
+	String link_t = p_type; // For links in metadata
+	String display_t = link_t; // For display purposes
 	if (is_enum_type) {
-		if (p_enum.get_slice_count(".") > 1) {
-			t = p_enum.get_slice(".", 1);
-		} else {
-			t = p_enum.get_slice(".", 0);
+		link_t = p_enum; // The link for enums is always the full enum description
+		Vector<String> slices = p_enum.split(".");
+
+		if (slices.size() == 0) { // This is a local enum with no class specifier
+			display_t = link_t;
+		} else { // There is a class specifier
+			// Replace :: from enum's use of fully qualified class names with regular .
+			slices.set(0, slices[0].replace("::", "."));
```

Comment:
Where is it coming from formatted like this? Can we ensure this doesn't happen?

And is it only affecting the first class in complex identifiers? Perhaps we can just run a replace on the original string?

Reply:
Wonderful comment, this allowed me to simplify the code a whole lot! It's way simpler now!!! Thank you!